apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/pac/AbstractR4ResourceProvider.java

```diff
@@ -65,9 +66,18 @@ public abstract class AbstractR4ResourceProvider<T extends IBaseResource>
 
   /**
    * A {@link Pattern} that will match the {@link ClaimResponse#getId()}s used in this application,
-   * e.g. <code>f-1234</code> or <code>m--1234</code> (for negative IDs).
+   * e.g. <code>f-1234</code> or <code>m--1234</code> (for negative IDs). e.g. <code>
+   * f-LTA0M2EyNWU2YjM0MmRmOTczY2YyYjU</code> or <code>m--00009127422efa</code> (for negative IDs).
+   *
+   * <p>Observation of data on ENVs, prod, prod-sbx, test: mcs_claim table, column idr_clm_hd_idn
+   * char(15): -00009127422efa, -100000103 fiss_claim table, column claim_id char(43):
+   * LTA0M2EyNWU2YjM0MmRmOTczY2YyYjU, MjIzMzYzMDQzOTIyMDdDQUFZICAgICAwMTAxMQ
    */
-  private static final Pattern CLAIM_ID_PATTERN = Pattern.compile("([fm])-(-?\\p{Digit}+)");
+  private static final Pattern CLAIM_ID_PATTERN =
+      Pattern.compile("^(f)-(\\p{Alnum}+)$|^(m)-(-?\\p{Alnum}+)$");
```

Comment:
Do we need to split the regex in two? Would this work (just changing {Digit} to {Alnum}? 
```suggestion
      Pattern.compile("^([fm])-(-?\\p{Alnum}+)$");
```

Reply:
Hmm. I suggested that this be split to honor only alphanumeric in the fiss case, and numeric for the mcs case, e.g. `^(f)-(-?\\p{Alnum}+)$|^(m)-(-?\\p{Digit}+)$`.

Reply:
That makes sense, this should be the change then
```suggestion
      Pattern.compile("^(f)-(\\p{Alnum}+)$|^(m)-(-?\\p{Digit}+)$");
```

Reply:
> That makes sense, this should be the change then

the mcs pattern needs to use {Alnum} since there are m typed claim id in mcs_claim tabel that contains alphabetic chars...

Reply:
Yeah, it seems the test data does have alphanumeric values even though in prod it's just numeric. Another example of problematic test data. In that case we can use my initial suggestion and avoid splitting out the regex.

Reply:
> Yeah, it seems the test data does have alphanumeric values even though in prod it's just numeric. Another example of problematic test data. In that case we can use my initial suggestion and avoid splitting out the regex.

yea, since the | separated 2 sub-regex patterns are almost identical..., can change to a 2 group capture...

Reply:
Oh sorry, I meant the data in prod-sbx